File: Estrutura_Condicional/C/Ex1983/main.c
#include <stdio.h>

int main() {
    int n, matricula, melhor_matricula;
    double nota, melhor_nota = -1.0;

    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%d %lf", &matricula, &nota);
        if (nota > melhor_nota) {
            melhor_nota = nota;
            melhor_matricula = matricula;
        }
    }

    if (melhor_nota >= 8.0)
        printf("%d\n", melhor_matricula);
    else
        printf("Minimum note not reached\n");

    return 0;
}

File: Estrutura_Condicional/Java/Ex1983/Main.java
package Ex1983;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int matricula, melhorMatricula = 0;
        double nota, melhorNota = -1.0;

        for (int i = 0; i < n; i++) {
            matricula = scanner.nextInt();
            nota = scanner.nextDouble();
            if (nota > melhorNota) {
                melhorNota = nota;
                melhorMatricula = matricula;
            }
        }

        if (melhorNota >= 8.0)
            System.out.println(melhorMatricula);
        else
            System.out.println("Minimum note not reached");
        
        scanner.close();
    }
}